c7n/resources/rds.py

```diff
@@ -423,3 +428,30 @@ def process_snapshot_set(self, snapshots_set):
                     DBSnapshotIdentifier=s['DBSnapshotIdentifier'])
             except ClientError as e:
                 raise
+
+
+class ARNGenerator(object):
+    """Base class for RDS ARN generators.
```

Comment:
this might be nicer as just a function in utils, with functools.partial to fill out for particular users/callers.